packages/client/src/views/RecordAudit/ActionMenu.tsx

```diff
@@ -0,0 +1,646 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+
+import React from 'react'
+import { DropdownMenu } from '@opencrvs/components/lib/Dropdown'
+import {
+  DangerButton,
+  PrimaryButton,
+  TertiaryButton
+} from '@opencrvs/components/lib/buttons'
+import { CaretDown } from '@opencrvs/components/lib/Icon/all-icons'
+import { useDispatch } from 'react-redux'
+import { Icon, ResponsiveModal } from '@opencrvs/components'
+import {
+  goToCertificateCorrection,
+  goToHome,
+  goToIssueCertificate,
+  goToPage,
+  goToPrintCertificate,
+  goToUserProfile,
+  goToViewRecordPage
+} from '@client/navigation'
+import { IntlShape } from 'react-intl'
+import { Scope } from '@sentry/react'
+import { IDeclarationData } from './utils'
+import {
+  clearCorrectionAndPrintChanges,
+  deleteDeclaration,
+  DOWNLOAD_STATUS,
+  IDeclaration,
+  SUBMISSION_STATUS,
+  unassignDeclaration
+} from '@client/declarations'
+import { CorrectionSection } from '@client/forms'
+import { buttonMessages } from '@client/i18n/messages'
+import { UserDetails } from '@client/utils/userUtils'
+import { EVENT_STATUS } from '@client/workqueue'
+import {
+  DRAFT_BIRTH_PARENT_FORM_PAGE,
+  DRAFT_DEATH_FORM_PAGE,
+  DRAFT_MARRIAGE_FORM_PAGE,
+  REVIEW_CORRECTION,
+  REVIEW_EVENT_PARENT_FORM_PAGE
+} from '@client/navigation/routes'
+import { messages } from '@client/i18n/messages/views/action'
+import { useModal } from '@client/hooks/useModal'
+import { DeleteModal } from '@client/views/RegisterForm/RegisterForm'
+import { client } from '@client/utils/apolloClient'
+import { Event } from '@client/utils/gateway'
+import { conflictsMessages } from '@client/i18n/messages/views/conflicts'
+import { GQLAssignmentData } from '@client/utils/gateway-deprecated-do-not-use'
+
+export const ActionMenu: React.FC<{
+  declaration: IDeclarationData
+  intl: IntlShape
+  scope: Scope
+  draft: IDeclaration | null
+  userDetails: UserDetails | null
+  duplicates?: string[]
+  toggleDisplayDialog: () => void
+  clearCorrectionAndPrintChanges: typeof clearCorrectionAndPrintChanges
+  goToPage: typeof goToPage
+  goToPrintCertificate: typeof goToPrintCertificate
+  goToIssueCertificate: typeof goToIssueCertificate
+  goToUserProfile: typeof goToUserProfile
+}> = ({
+  declaration,
+  intl,
+  scope,
+  draft,
+  userDetails,
```

Comment:
Generally: let's not prop drill `intl` or other objects, as React is bad at diffing them and usually it makes purer and cleaner components by just passing the bare minimum to them what they need. If you need intl- here, use the `useIntl` hook.

I think this also applies to the goToFoo -functions, we can just import them directly without passing them as props and then dispatching them.